src/main/kotlin/systems/ajax/malov/stockanalyzer/config/beanpostprocessor/LogExecutionTimeAnnotationBeanPostProcessor.kt

```diff
@@ -0,0 +1,86 @@
+package systems.ajax.malov.stockanalyzer.config.beanpostprocessor
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import java.lang.reflect.Method
+import java.lang.reflect.Proxy
+
+
+@ConditionalOnProperty(name = ["logExecutionTime.enabled"], havingValue = "true")
+@Component
+class LogExecutionTimeAnnotationBeanPostProcessor : BeanPostProcessor {
+
+    private val beanMap = HashMap<String, ClassWithAnnotatedMethods>()
+
+    data class ClassWithAnnotatedMethods(val clazz: Class<*>, val methods: List<Method>)
```

Comment:
should it be private?

----

I'd like to see an internal classes after all members of the host class